src/child/containers/toolbar/Toolbar.js

```diff
@@ -37,10 +37,12 @@ class Toolbar extends Component {
     }
 
     onCompactClick() {
+        fin.desktop.Window.getCurrent().resizeTo(230, 500, 'top-right');
         this.props.dispatch(compact());
```

Comment:
Unsure about this, so please correct me if I'm wrong, but it seems odd to resize the window, then dispatch an action. It feels like the resize is a side-effect.

I think using an async action creator or subscribing a listener on the store to call the resize method would be more appropriate.

I'm not sure which, but I'm leaning towards an async action creator. Assuming the `compact` and `expand` actions will always lead to a resize and as `resizeTo` is aysnc and will either succeed or fail, the async action creator seems like a good fit. For example, at the moment if `resizeTo` failed in `onCompactClick()` the state would still be updated to suggest that the window is in compact view, when it's not.

Any thoughts?